lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -238,6 +255,49 @@ void StructFieldAccessManager::buildCloseProximityRelations() {
   }
 }
 
+void StructFieldAccessManager::suggestFieldReordering(bool UseOld) {
+  outs() << "------------- Suggestions on Reordering ------------------\n";
+  for (auto &it : CloseProximityBuilderMap) {
+    auto *type = it.first;
+    if (type->isLiteral()) {
+      outs() << "Recommendation on an anonymous struct ";
+    } else {
+      outs() << "Recommendation on struct [" << type->getStructName() << "] ";
+    }
+    auto Hotness = HotnessAnalyzer->getHotness(type);
+    assert(Hotness);
+    outs() << " (Hotness "
+           << 100 * Hotness.getValue() / HotnessAnalyzer->getMaxHotness()
+           << "% of hottest struct):\n";
+    auto *FRA =
+        UseOld
+            ? new OldFieldReorderAnalyzer(CurrentModule, type, it.second, NULL)
+            : new FieldReorderAnalyzer(CurrentModule, type, it.second, NULL);
+    FRA->makeSuggestions();
```

Comment:
Similar to comment below - don't dynamically alloc the struct. Cleaner to have a if(UseOld) {OldFireldReorderanalyuzer OFRA; OFRA.makeSuggestions}...